server/src/main/java/org/opensearch/index/IndexService.java

```diff
@@ -497,10 +504,23 @@ public synchronized IndexShard createShard(
                 }
             };
             Directory directory = directoryFactory.newDirectory(this.indexSettings, path);
+            Directory remoteDirectory = null;
+            if (this.indexSettings.isRemoteStoreEnabled()) {
+                try {
+                    Repository repository = repositoriesService.repository("dragon-stone");
+                    remoteDirectory = remoteDirectoryFactory.newDirectory(this.indexSettings, path, repository);
+                } catch (RepositoryMissingException e) {
+                    throw new IllegalArgumentException(
+                        "Repository should be created before creating index with remote_store enabled setting",
```

Comment:
Is repository created by user or by system?

Reply:
As of now, it is created by user. Once we make the name configurable as a part of https://github.com/opensearch-project/OpenSearch/issues/3286, we can check if the repository can be created during bootstrap.

Reply:
So, you're saying as of now it will have to be created by user but he can only name it as "dragon-stone"? That doesn't seem correct right?